apps/zui/src/views/results-pane/run-results-query.tsx

```diff
@@ -15,7 +15,7 @@ export const runResultsMain = createHandler(
 
     // See if we can paginate this query
     await waitForSelector(QueryInfo.getIsParsed).toReturn(true)
-    const canPaginate = !select(QueryInfo.getIsSummarized)
+    const canPaginate = select(QueryInfo.hasAggregation)
```

Comment:
@nwt: I might be able to help field this one. As described in #2748 and https://github.com/brimdata/zed/issues/4491, it's my understanding that Zui's always taken the conservative stance of not trying to do pagination on aggregations because they have the potential to be expensive and hence considered a bad idea to keep re-executing the whole thing just to get single pages of results. By contrast with straight search results the backend has benefit of the the pool key and other potential optimizations, hence the app can ask for page-sized subsets via `head`/`tail` without putting excess burden on the backend.

Reply:
@philrz: As I understand it, this pagination decision is about correctness rather than performance. The `summarize` operator produces output with nondeterministic order, so you can't paginate that output correctly by re-executing unless you sort it before applying `head` and `tail`.

In fact, now that I've thought about it, I don't think we can paginate `summarize` output correctly unless it's sorted by all of the `summarize` keys.

Reply:
@nwt: Yes, that all makes sense. But I think both can be true. Without digging up years of history to confirm, I think I remember a time when our aggregations did return results in a deterministic order, and so the decision to not try to paginate aggregation results was at one time based on the performance argument as recorded in those issues (I didn't make that stuff up... I recall it being the outcome of chats with others.) So even if we put in some kind of logic to confirm if the results were sorted, the perf argument might remain.

Reply:
That's what I thought but @philrz noticed that the behavior changed when this was based off of isSorted. I think basing pagination on whether or not the query is sorted is the right way to go but this change should be done on a separate pr and we should try to keep things as close to the way they were in this pr.

Reply:
@mattnibs: That works for me.